x-pack/plugins/fleet/common/services/agent_policies_helpers.ts

```diff
@@ -5,8 +5,19 @@
  * 2.0.
  */
 
-import type { AgentPolicy } from '../types';
-import { FLEET_SERVER_PACKAGE, FLEET_APM_PACKAGE, FLEET_SYNTHETICS_PACKAGE } from '../constants';
+import type { NewAgentPolicy, AgentPolicy } from '../types';
+import {
+  FLEET_SERVER_PACKAGE,
+  FLEET_APM_PACKAGE,
+  FLEET_SYNTHETICS_PACKAGE,
+  FLEET_ENDPOINT_PACKAGE,
+} from '../constants';
+
+export function isAgentPolicy(
+  policy: Partial<NewAgentPolicy | AgentPolicy>
+): policy is AgentPolicy {
+  return (policy as AgentPolicy).is_protected !== undefined;
```

Comment:
Can we use other more specific prop in Agent policy to check if it is an agent policy or not?

Reply:
do you think `revision` in `AgentPolicy` works better maybe?